airflow/compat/hashlib_wrapper.py

```diff
@@ -0,0 +1,45 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import hashlib
+import sys
+from typing import Any
+
+# Check if "usedforsecurity" is available for hashlib
+if sys.version_info >= (3, 9):
+    HAS_USEDFORSECURITY = True
+else:
+    HAS_USEDFORSECURITY = False
+
+
+def md5(data: Any, used_for_security: bool | None = None):
```

Comment:
I feel we should match the original signature as much as possible, namely

1. Match the `usedforsecurity` argument name
2. Name this module `hashlib` (removing the `_wrapper` part)

Reply:
Also I think `data` should not be `Any`? The stdlib documentation seems to indicate it must be `bytes`.

Reply:
Ah, I can definitely update the typing.
To your 1st point though, it feels wrong to give it the same name as an existing stdlib module. I'd rather be explicit with what we are trying to do. Also, used_for_security felt more pythonic and easier to read. But I am by no means a python expert, so I could be swayed.

Reply:
I'd rename the `usedforsecurity` but wouldn't remove the `_wrapper` (or rather rename `_wrapper` to `_compat` if the wrapper name is not nice.  I think this is usually the approach we take and I think there is a value in making sure that we know it's not "the" hashlib library. Might avoid some surprises (you woll only see the difference if you scroll up to the imports) and we might not forget to remove it when we bump min-airflow version to 3.9 (which is still long time from now)

Reply:
I have made these suggested changes, leaving `_wrapper` at the end.

Reply:
I'd personally like `_compat` over `_wrapper`.

Fwiw, this isn't our convention (at least looking at existing compat stuff). We have `functools`, not `functools_compat`.

I'd rather see
```
from airflow.compat.hashlib import md5
md5(...)
```
than
```
from airflow.compat import hashlib_compat
hashlib_compat.md5(...)
```

My 2c.

Reply:
Naming a submodule with the same name as a top-level module is not a problem (not in Python 3 anyway) since `airflow.compat.hashlib` and `hashlib` can never conflict. The `_wrapper` or `_compat` suffix is simply redundant. If you really feel strongly about having a suffix (this particular case is slightly different from `functools` since the `md5` function is a wrapper, not a shim), you can choose to put the module in `airflow.utils` or even directly under `airflow` (a la `airflow.typing_compat`) instead.

Reply:
Sorry, I thought I responded. I moved things under `utils`.